### ISSUE TITLE
G4/G12 - Active menu items (on local nav + sidebar menus) have different styling, but do not communicate to screen-reader that they're the active page

### ISSUE THREAD

**Issue by mmafe** [brharp/hjckrrh#190]
Active menu items have different styling, but do not communicate to screen-reader that they're the active page. For future reference when solving this: http://www.heydonworks.com/article/the-accessible-current-page-link-conundrum

**Comment by mmafe** [comment]
Fixed on G4 local nav.

**Comment by mmafe** [comment]
Fixed on sidebar menu blocks as well (please include in testing)

**Comment by jpinsent** [comment]
The fix works, but there's an odd quirk that I don't think is intended. When you listen to the currently selected menu item, the screen reader just reads the link, i.e. "FAQ visited link". When you try to move to the next item on the page however, it then reads "(current page)". I feel like the screen reader should announce "current page" at the same time it reads the menu link.

Tested with NVDA in Chrome and Firefox and both function the same way.

**Comment by jpinsent** [comment]
Also, this is just a suggestion, but would it be possible to highlight the currently selected menu item on the local nav a little more? It appears to have a very slight contrast increase compared to the other menu items, but it is not very noticeable.

**Comment by mmafe** [comment]
**Update:** Moved the aria-describedby label into the link. However, it looks like the (current page) aria-describedby label is only read out when NVDA is in Focus/Forms mode (i.e. tabbing) instead of Browse/Reading mode (i.e. using arrow keys). This seems to be true for any aria-describedby value read out by NVDA.

To fix it so folks hear the label in both browse and focus mode, I'm going to add the "(current page)" label to the link, but make it visible to screen-readers only instead of hidden to all. I'm also removing the aria-describedby connection, otherwise the link would then be reading the label twice when in focus mode (i.e. Name of link (current page), link, (current page)).

**Comment by mmafe** [comment]
**New (Branching) Issue:** Increasing content for currently selected menu in G4 (local nav) now being tracked in https://github.com/brharp/hjckrrh/issues/201

**Comment by jpinsent** [comment]
Seems like it's working as intended. There's just one small problem; when you're on a page within a submenu item on the side menu, the screen reader will only read '(current page)' with the submenu item. The parent menu item is highlighted, but the screen reader will read it normally. I'd suggest removing the highlighted text theming on the parent item.

![problem](https://cloud.githubusercontent.com/assets/12450480/9669995/3cad23d4-5258-11e5-804a-856f1c866164.PNG)

**Comment by mmafe** [comment]
Assistive tech can still pick up that the (highlighted) parent menu item is the parent list of the sub-menu list. Because of that, I think we're okay to leave the highlighted text theming since the same information is reflected in the (nested list) structure of the menu.
